Created a dummy 'variable' function in the numpy backend to replace the `BACKEND` variable.

gabrieldemarmiesse opened the pull request:
### Summary
@taehoonlee 

Allow me to propose an alternative to the `assert_value_with_ref` method for the numpy reference operations.
The limitation which I see with this approach is that it will be hard to track the progress of the removal of the other backends (I may be wrong, feel free to correct me it's possible I haven't read the code well enough).

I propose that we consider the numpy backend as a true backend for the tests (without gradients of course).
I created a new variable called `BACKEND_AND_NP` which is similar to `BACKENDS` and will hopefully replace it.

Once we manage to remove all occurences of `KC`, `KTF`, `KTH` and `BACKENDS`, we should be able to removed unnecessary installs of backends.

I made a small example here for the `reshape` backend function.

What are your thoughts on this?

### Related Issues

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

taehoonlee commented:
That's what I wanted to do. LGTM.

taehoonlee commented on tests/keras/backend/backend_test.py:
```diff
@@ -12,6 +12,7 @@
 
 
 BACKENDS = []  # Holds a list of all available back-ends
+BACKEND_AND_NP = [K, KNP]
```
How about just `WITH_NP`?

gabrieldemarmiesse replied:
I'm really bad at naming variables, haha. Maybe `K_AND_KNP` ? That would be the most explicit.

taehoonlee replied:
You are such a big help :) But I think both `K` and `WITH_NP` are implicit. And `check_single_tensor_operation(op, ..., WITH_NP, ...)` is more natural to read than `K_AND_KNP`.

gabrieldemarmiesse replied:
Thanks for your kind words. Let's go with `WITH_NP` then.

gabrieldemarmiesse commented:
Since I had a bit of time to put into it, I moved the tests which don't run with CNTK to new test functions. I also replaced BACKENDS by WITH_NP where it was possible.
